src/lib/pages/code-details/index.tsx

```diff
@@ -144,17 +145,22 @@ const CodeDetailsBody = observer(
 const CodeDetails = observer(() => {
   useWasmConfig({ shouldRedirect: true });
   const router = useRouter();
-  const codeIdParam = getFirstQueryParam(router.query.codeId);
-  const data = useCodeData(codeIdParam);
+
+  const validated = zCodeDetailsQueryParams.safeParse(router.query);
+
+  if (!validated.success) {
+    return <InvalidCode />;
```

Comment:
Move this condition under the `PageContainer` component.

Reply:
👍🏼